README.md

```diff
@@ -0,0 +1,93 @@
+  EKS cluster (including all network and IAM dependencies).
+* `fluentbits.charts.cnp.example.org` - Defines a composite resource
+  representing a Helm release of the Fluent Bit logging daemonset.
+* `clusters.cnp.example.org` - Defines a composite resource representing a 'CNP
+  Kubernetes Cluster'. This cluster is composed of the EKS cluster and Fluent
+  Bit composite resources above, and thus this package depends on the above two.
+  **The files in this package contain explanatory annotations; it's the most
+  interesting one to look at.**
+
+The root of each package directory contains a file named `crossplane.yaml`. This
+file is required. A Crossplane package is an OCI image that is expected to emit
+this file, followed by any packaged resources in the order they should be
+applied, when its entrypoint is invoked.
+
+To create a package, the package author (or their CI/CD system) runs the
+following commands:
+
+```bash
+# Any OCI image that omits a YAML resource of kind: Configuration or kind:
+# Provider (in the pkg.crossplane.io API Group) followed by zero or more
+# supplemental resources when its ENTRYPOINT is invoked is considered a valid
+# Crossplane package. This tools builds a valid package by reading a file named
+# crossplane.yaml at the root of the working directory (which must contain a
+# kind: Configuration or a kind: Provider), then discovering all Crossplane
+# resources in or under the working directory. These resources are concatenated
+# into a YAML stream in a randomly generated filename (to avoid implementations
+# relying on the path of this file), and the ENTRYPOINT of the OCI image is set
+# to `cat randomlynamedstream.yaml`.
+$ crossplane package build --tag exampleorg/cnp-cluster:v0.1.0
+
+# The `crossplane package` command provides a similar interface to the `docker`
+# tool frequently used to build, tag, and push images. `crossplane package`
+# publishes packages to the Upbound Cloud Registry rather than the Docker Hub
+# by default. These tools are recommended but optional; you could package a
+# Python script that emitted a valid stream of YAML if you wanted to.
+$ crossplane package push exampleorg/cnp-cluster:v0.1.0
+```
+
+Note that package metadata does not include a version. Crossplane packages use
+the package (image) tag as their version, and thus require that tags be valid
+semantic versions. Images built by CI/CD as part of a gitops flow are encouraged
+to derive their semantic version from a git tag. In such a case all packages
+defined in this repository would be versioned in lock-step.
+
+## Proposed Changes
+
+This sketch assumes the acceptance of the in-flight [Package Manager refactor].
+It also:
+
+* Renames `InfrastructureDefinition` to `CompositeResourceDefinition` (XRD).
+* Replaces `InfrastructurePublication` with a configuration field on an XRD.
+* Automatically derives UI metadata from the OpenAPI schema of an XRD.
```

Comment:
Are we sure that this is the way we want to handle UI metadata going forward?
Doesn't this conflict with the existence of `clusters.cnp.example.org/.crossplane/allocatedcluster.ui-schema.yaml`?

Reply:
We should be able to generate sane defaults for UI if there is no ui-schema but generation can't replace things like ObjectRef(which object type, and what's the format of the payload?).

Reply:
yes, based on previous discussion we had 3 phases:

1. ui meta as currently defined (but under the new structure under the `.crossplane` directory)
1. support OpenAPI schema as a graceful fallback - i.e. if no ui meta was provided
1. deprecate ui meta fields that are overlapping with OpenAPI schema

for phase 1, we'd be using the `.crossplane` directory of UI meta (e.g. as defined in the PR for `AllocatedCluster`), with the intent that since the published resource `Cluster` has the same schema, that the UI meta for `AllocatedCluster` could also be used for the published `Cluster` resource so we didn't need to include duplicate UI meta files.

then over time we could move to phase 2/3.

Reply:
Tracked in https://github.com/crossplane/example-cnp/issues/5.